hook!-full-game/src/Player/States/Wall.gd

```diff
@@ -0,0 +1,39 @@
+extends State
+
+const WALL_SLIDE_SPEED: = 100.0
+const WALL_PULL_STRENGTH: = 300.0
+
+var _jump_strength: = 1000.0
+var _velocity: = Vector2.ZERO
+var _wall_direction: = -1
+
+func enter(msg: Dictionary = {}) -> void:
+	_wall_direction = msg.direction
+
+
+func physics_process(delta: float) -> void:
+	_velocity.y += WALL_SLIDE_SPEED * delta
+	_velocity.x += (WALL_PULL_STRENGTH * -_wall_direction) * delta
+	_velocity = _player.move_and_slide(_velocity, _player.FLOOR_NORMAL)
```

Comment:
The friction of the wall prevents WALL_SLIDE_SPEED from having any impact on the motion. Right now the slide is too slow compared to the overall rhythm of the game. I tried to only update _velocity.x but it seems the issue comes from move_and_slide?

Reply:
I tried to instead use the lerp function so we maintain the falling momentum for a bit, dropping it down to the `max_slide_speed`. I exported a value so we can tweak the time it takes for this to happen, something around 0.07 and 0.15 looked nice to me, try to test it out :D